types/ramda/test/map-tests.ts

```diff
@@ -36,9 +36,23 @@ import * as R from 'ramda';
     b: string;
   }
 
+  interface C {
+    b: number;
+    c: string;
+  }
+
   R.map<A, A>(R.inc, { a: 1, b: 2 });
   R.map<A, B>(R.toString, { a: 1, b: 2 });
 
   R.map<A, A>(R.inc)({ a: 1, b: 2 });
   R.map<A, B>(R.toString)({ a: 1, b: 2 });
+
+  type KeyOfUnion<T> = T extends infer U ? keyof U : never;
+
+  // $ExpectType Record<"a" | "b" | "c", void>
+  R.map<A | C, Record<KeyOfUnion<A | C>, void>>(
+    // $ExpectType (value: string | number) => void
```

Comment:
I don't think it's necessary to test the types and return value of a static arrow function written for a test.